test/framework/src/main/java/org/elasticsearch/cluster/metadata/DataStreamTestHelper.java

```diff
@@ -571,6 +571,7 @@ public static IndicesService mockIndicesServices(MappingLookup mappingLookup) th
             DocumentMapper documentMapper = mock(DocumentMapper.class);
             when(documentMapper.mapping()).thenReturn(mapping);
             when(documentMapper.mappingSource()).thenReturn(mapping.toCompressedXContent());
+            when(documentMapper.mappers()).thenReturn(mappingLookup);
```

Comment:
It turns out we were sometimes getting NPEs when calling `DocumentMapper#mappers` due to an incomplete mock here 🙃